Add IdentityX newsletter components

B77Mills opened the pull request:
**REQUIRES: https://github.com/parameter1/identity-x/pull/16**
 - Add ability to set successMessageType(login || newsletter-signup)
 - Add ability to send forceProfileReVerification in additionalEventData which will force the user to submit their profile on next login
<img width="1155" alt="Screen Shot 2022-06-21 at 2 02 23 PM" src="https://user-images.githubusercontent.com/3845869/174878925-bc734327-f55a-4775-a36b-58dde40bbcc3.png">
<img width="1134" alt="Screen Shot 2022-06-21 at 2 02 38 PM" src="https://user-images.githubusercontent.com/3845869/174878929-5d6f44ab-e912-4769-8cdd-3b02e7babb42.png">

B77Mills committed:
Add ability to set success message type

Add the ability to set the successMessageType on login component.  The valid options for now are login & newsletter-signup

The default is login.

Also added missing component prop(additionalEventData).

B77Mills committed:
Add forceProfileReVerification suport to login route

Add the ability to send forceProfileReVerification in the addintionalEventData.  When set it will set the user.forceProfileReVerification to true making the user have to resubmit their profile on next login.

B77Mills committed:
correctly set v-html with no internal prop call

solocommand committed:
Fix button labels, action text

solocommand committed:
Move translation and newsletter state to monorail

solocommand committed:
Cleanup example site newsletters

solocommand committed:
Add new IdX Vue components

solocommand committed:
Add IdX enl marko components

solocommand committed:
Support custom newsletter block in footer

solocommand committed:
Add modifiers for signup forms

solocommand committed:
Add comments/etc to example site

solocommand committed:
Swap to use new IdX newsletter components

solocommand committed:
Fix breakpoint

solocommand committed:
Add social, fix padding/titles

solocommand commented:
Updated implementation allows for reusability by moving components into Monorail theme, with backwards compatibility for RR (new components, existing ones stay as-is). To implement, use the new `identity-x-newsletter-form-inline`, `identity-x-newsletter-form-footer` (along with new `@newsletter-block` attribute for theme footer) and `identity-x-newsletter-form-pushdown` (in place of "theme-site-newsletter-menu").

|Initial|Submitted|Logged in |
|-|-|-|
| ![screencapture-www-p1-sandbox-dev-parameter1-51269-directory-news-22288695-lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-2022-06-27-12_52_28](https://user-images.githubusercontent.com/1778222/176005731-274c0ce5-28c0-4ddf-96fb-969c38b4a8c3.png) | ![screencapture-www-p1-sandbox-dev-parameter1-51269-directory-news-22288695-lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-2022-06-27-12_53_27](https://user-images.githubusercontent.com/1778222/176005721-d72d3e5e-e29a-4b44-971a-f212c800047b.png) | ![screencapture-www-p1-sandbox-dev-parameter1-51269-directory-news-22288695-lorem-ipsum-dolor-sit-amet-consectetur-adipiscing-elit-2022-06-27-12_54_12](https://user-images.githubusercontent.com/1778222/176005697-0aff3182-50f8-4338-a5da-1218af80080c.png) |

B77Mills committed:
remove custom event handling for new idx newsletter comps.

B77Mills committed:
Add new idx-newsletter-toggle button.

This allows for both versions(idx & non-idx) of the newsletter components to work independently.  We will have to work on adding the missing reporting to the new idx components as phase 2.

B77Mills committed:
css change to allog banner to span full width & contents centered

B77Mills commented:
@solocommand  my last two commits I added a idx-newsletter-toggle component that works with the idx-pushdown component.  This allows for everything to be independent from the current newsletter stuff.  I also adjusted the css for the inline component.

**Screen shots of them loaded side by side:**
 - **Pushdown** <img width="1341" alt="Screen Shot 2022-06-28 at 9 51 11 AM" src="https://user-images.githubusercontent.com/3845869/176210438-9f75d69d-5f21-4423-9814-69b14296feab.png">
 - **Content Inline** <img width="978" alt="Screen Shot 2022-06-28 at 9 50 59 AM" src="https://user-images.githubusercontent.com/3845869/176210610-db7260e5-2a5e-4828-b623-0dbd3099d75a.png">
 - **Section Inline** <img width="1180" alt="Screen Shot 2022-06-28 at 9 51 26 AM" src="https://user-images.githubusercontent.com/3845869/176210715-e98ef325-7a15-45c4-8baf-9185a2276e28.png">

B77Mills committed:
Add useIdxNewsletterSignup support to footer/header

Add useIdxNewsletterSignup boolean prop to determine which toggle button & footer signup block to use

B77Mills committed:
Set example site to use theme-site-header component

B77Mills committed:
move pushdown component into site-header & add disabled configuration

solocommand commented on packages/marko-web-theme-monorail/components/site-header.marko:
```diff
@@ -62,3 +63,12 @@ $ const navigation = {
+    <else>
+      <theme-site-newsletter-menu />
+    </else>
+</marko-web-block>
```
@B77Mills Indentation is off here. Is there a reason you're wrapping this in a marko-web-block tag? That doesn't seem to be done within the site-specific implementations...would just remove it unless it's necessary for some reason.

B77Mills replied:
@solocommand  Indentation is fixed and i removed the block wrapper.

B77Mills committed:
Fix Indentation

B77Mills committed:
Remove marko-web-block element
